docs/standard/datetime/snippets/timezone-concepts/TimeZone2Concepts.cs

```diff
@@ -0,0 +1,389 @@
+﻿using System.Collections.ObjectModel;
+using System.Security;
+
+[assembly: CLSCompliant(true)]
+
+public class TZExamples
+{
+    public static void Main()
+    {
+        // tze.IterateTimeZones();
+        // tze.SelectTimeZone();
+        ShowDaylightStatus();
+        Console.WriteLine("\nShowLocalAndUtcTime:");
+        TZExamples.ShowLocalAndUtcTime();
+        TZExamples.ConvertToArbitraryTime();
+        Console.WriteLine("**ConvertTimeToUtc***");
+        TZExamples.ConvertToUtc();
+        Console.WriteLine("ConvertEasternToUtc:");
+        TZExamples.ConvertEasternToUtc();
+        Console.WriteLine("\nConvertUtcToCentral:");
+        ConvertUtcToCentral();
+        Console.WriteLine("\nConvertHawaiianToLocal:");
+        ConvertHawaiianToLocal();
+        Console.WriteLine("Resolving ambiguous times:");
+        Console.WriteLine(TZExamples.ResolveAmbiguousTime(new DateTime(2006, 10, 29, 02, 03, 15)));
+        Console.WriteLine(TZExamples.ResolveAmbiguousTime(DateTime.Now));
+        Console.WriteLine();
+
+        TZExamples tze = new();
+        tze.GetUserDateInput();
+    }
+
+    private static void IterateTimeZones()
+    {
+        // <Snippet1>
+        ReadOnlyCollection<TimeZoneInfo> tzCollection;
+        tzCollection = TimeZoneInfo.GetSystemTimeZones();
+        // </Snippet1>
+
+        Console.WriteLine($"Listing {tzCollection.Count} time zones found on the system:");
+        // <Snippet12>
+        foreach (TimeZoneInfo timeZone in tzCollection)
+            Console.WriteLine($"   {timeZone.Id}: {timeZone.DisplayName}");
+        // </Snippet12>
+    }
+
+    private static void SelectTimeZone()
+    {
+        TZListForm frm = new TZListForm();
+        frm.ShowDialog();
+    }
+
+    private static void ShowDaylightStatus()
+    {
+        // <Snippet3>
+        DateTime dateToday = DateTime.Now;
+        TimeSpan differenceFromUtc = TimeZoneInfo.Local.GetUtcOffset(dateToday);
+        Console.WriteLine("The time is {0:t} in {1} time, {2:##.0} hours {3} universal time.",
+                          dateToday,
+                          TimeZoneInfo.Local.IsDaylightSavingTime(dateToday) ? "daylight saving" : "standard",
+                          Math.Abs(differenceFromUtc.TotalHours),
+                          differenceFromUtc.Hours > 0 ? "after" : "earlier than");
```

Comment:
An example like this *could* use interpolated strings, but I'm ambivalent. You could add new lines in the interpolated expressions, but that often looks clumsy. I've only used that syntax with LINQ queries embedded in string message. If you wanted to use it, it would be something like this:

```suggestion
        Console.WriteLine($"The time is {dateToday:t} in {(TimeZoneInfo.Local.IsDaylightSavingTime(dateToday) ? 
            "daylight saving" : "standard")} time, {Math.Abs(differenceFromUtc.TotalHours):##.0} hours {(differenceFromUtc.Hours > 0 
            ? "after" : "earlier than")} universal time.");
```